.github/workflows/ci.yml

```diff
@@ -586,26 +663,62 @@ jobs:
         if: failure()
         uses: andymckay/cancel-action@0.5
 
-  omni-executor-check:
+  omni-executor-build:
     runs-on: ubuntu-latest
     needs:
       - fmt
       - set-condition
       - sequentialise
     steps:
       - uses: actions/checkout@v4
-      - name: Clippy
-        if: needs.set-condition.outputs.check_omni_executor == 'true'
-        working-directory: ./tee-worker/omni-executor
-        run: cargo clippy --release -- -D warnings
-      - name: Build
-        if: needs.set-condition.outputs.check_omni_executor == 'true'
-        working-directory: ./tee-worker/omni-executor
-        run: cargo build
-      - name: Tests
-        if: needs.set-condition.outputs.check_omni_executor == 'true'
-        working-directory: ./tee-worker/omni-executor
-        run: cargo test
+
+      - name: Free up disk space
+        if: startsWith(runner.name, 'GitHub Actions')
+        uses: ./.github/actions/disk-cleanup
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          # use the docker driver to access the local image
+          # we don't need external caches or multi platforms here
+          # see https://docs.docker.com/build/drivers/
+          driver: docker
+
+      - name: Build local builder
```

Comment:
Maybe we don't need this intermediate image while still having two stages in Dockerfile

It's built for other two workers for cache purpose

Reply:
I think we still need two stages. Actually I tried to merge them into one step. But the `builder` stage uses `rust:1.82-bookworm`, which is quite big. The final image pushed is also very big. That's the reason I revert back to two stages.

Reply:
We still have two stages in Dockerfile, but we don't need to build a `local-builder` first IMO

Reply:
I have removed the `local-builder`.